pkg/services/sqlstore/team.go

```diff
@@ -24,26 +25,58 @@ func init() {
 	bus.AddHandler("sql", IsAdminOfTeams)
 }
 
-func getTeamSearchSqlBase() string {
+func getFilteredUsers(signedInUser *models.SignedInUser) ([]int64, error) {
+	filteredUsers := make([]int64, 0)
+	if signedInUser == nil || signedInUser.IsGrafanaAdmin {
+		return filteredUsers, nil
+	}
+
+	ids, err := GetHiddenUsersIDs()
```

Comment:
this to me feels bad to have to issue a separate DB call here, can the query read user login so we can filter based that same as we do elsewhere why do we need to filter based on id here.

Reply:
I updated this, it adds a INNER JOIN in the query but it shouldn't be too heavy as it's restricted to the team members.